tokio/tests/task_local.rs

```diff
@@ -31,3 +31,16 @@ async fn local() {
     j2.await.unwrap();
     j3.await.unwrap();
 }
+
+#[cfg(tokio_unstable)]
+#[tokio::test(flavor = "current_thread")]
+async fn test_task_id() {
+    use tokio::task;
+
+    let handle = tokio::spawn(async {
+        let id = task::current_id();
+        println!("current task id: {}", id);
+    });
+
+    handle.await.unwrap();
+}
```

Comment:
we should probably have some tests that two or more tasks don't get colliding IDs.